hudi-common/src/main/java/org/apache/hudi/common/table/view/HoodieTableFileSystemView.java

```diff
@@ -398,14 +398,24 @@ protected Option<Pair<String, CompactionOperation>> getPendingLogCompactionOpera
 
   @Override
   protected boolean isPartitionAvailableInStore(String partitionPath) {
-    return partitionToFileGroupsMap.containsKey(partitionPath);
+    try {
+      readLock.lock();
+      return partitionToFileGroupsMap.containsKey(partitionPath);
+    } finally {
+      readLock.unlock();
+    }
   }
 
   @Override
   protected void storePartitionView(String partitionPath, List<HoodieFileGroup> fileGroups) {
-    LOG.debug("Adding file-groups for partition :" + partitionPath + ", #FileGroups=" + fileGroups.size());
-    List<HoodieFileGroup> newList = new ArrayList<>(fileGroups);
-    partitionToFileGroupsMap.put(partitionPath, newList);
+    try {
```

Comment:
This is where the map is updated but previously there was no write-lock used

Reply:
can you share with the us the incokation chain?

Reply:
`addFilesToView` and `ensurePartitionLoadedCorrectly` will update the store which is a map or spillable map for the most part. These were previously using a `read` lock instead of a `write`. This lead to the ability to update the map during a `read` which lead to the break in thread-safety.